### PR TITLE
Tournaments: add auto start feature

### PR DESCRIPTION
**Pull request by sirDonovan**

### CODE REVIEW

**Comment by ascriptmaster** [comment]
But what if the timeout parameter is negative?

**Comment by sirDonovan** [comment]
Good catch, added a minimum timeout.

File: tournaments/index.js [hunk 1]
```diff
@@ -493,6 +497,28 @@ Tournament = (function () {
...
+		if (this.isTournamentStarted) {
+			output.sendReply('|tournament|error|AlreadyStarted');
+			return false;
+		}
+		if (timeout < AUTO_START_MINIMUM_TIMEOUT || isNaN(timeout)) {
+			output.sendReply('|tournament|error|InvalidAutoStartTimeout');
+			return false;
+		}
+
+		timeout = parseFloat(timeout);
```
Comment (ascriptmaster) [inline tournaments/index.js]: Why call `parseFloat` after comparing the `AUTO_START_MINIMUM_TIMEOUT` instead of before?

Comment (sirDonovan) [inline tournaments/index.js]: What's the advantage of calling it before? The autodq function does the same so I didn't think much of it.

Comment (ascriptmaster) [inline tournaments/index.js]: Oh, that's right, Javascript typecasting exists (even if it's disgusting)

It was more on the readability, so you show that it's going to be a `float` before you compare.

Heck, if it's set up to always make it either a number or `NaN` I wonder if you even need to call `parseFloat` at all

Comment (Zarel) [inline tournaments/index.js]: Yeah, I'd do it before for readability.

Comment (sirDonovan) [inline tournaments/index.js]: Alright, updated the code.

**Comment by sirDonovan** [comment]
Any other changes needed here?

File: tournaments/index.js [hunk 2]
```diff
@@ -493,6 +497,28 @@ Tournament = (function () {
...
+		}
+		timeout = parseFloat(timeout);
+		if (timeout < AUTO_START_MINIMUM_TIMEOUT || isNaN(timeout)) {
+			output.sendReply('|tournament|error|InvalidAutoStartTimeout');
+			return false;
+		}
+
+		if (this.autoStartTimeout) clearTimeout(this.autoStartTimeout);
+		if (timeout === Infinity) {
+			this.room.add('|tournament|autostart|off');
```
Comment (Zarel) [inline tournaments/index.js]: Does this require a client change?

Comment (sirDonovan) [inline tournaments/index.js]: Yes, I have that ready.

**Pull request merged**